code/modules/reagents/chemistry/recipes/pyrotechnics.dm

```diff
@@ -42,14 +42,82 @@
 	id = "nitroglycerin_explosion"
 	required_reagents = list(/datum/reagent/nitroglycerin = 1)
 	required_temp = 474
+	strengthdiv = 1.5
+
+/datum/chemical_reaction/reagent_explosion/rdx
+	name = "RDX"
+	id = /datum/reagent/rdx
+	results = list(/datum/reagent/rdx= 2)
+	required_reagents = list(/datum/reagent/hexamine = 2, /datum/reagent/toxin/acid/nitracid = 1, /datum/reagent/acetone_oxide = 1 )
+	required_temp = 459
+	strengthdiv = 5
+
+/datum/chemical_reaction/reagent_explosion/rdx/on_reaction(datum/reagents/holder, created_volume)
+	if(holder.has_reagent(/datum/reagent/stabilizing_agent))
+		return
+	holder.remove_reagent(/datum/reagent/rdx, created_volume*2)
+	..()
+
+/datum/chemical_reaction/reagent_explosion/rdx_explosion
+	name = "RDX explosion"
+	id = "rdx_explosion"
+	required_reagents = list(/datum/reagent/rdx = 1)
+	required_temp = 474
+	strengthdiv = 4
+
+/datum/chemical_reaction/reagent_explosion/tatp/New()
+	SSticker.OnRoundstart(CALLBACK(src,.proc/UpdateInfo))
+
+/datum/chemical_reaction/reagent_explosion/tatp_explosion/New()
+	SSticker.OnRoundstart(CALLBACK(src,.proc/UpdateInfo))
```

Comment:
Why is this over SSticker? Just do this the normal way without any callbacks or other fuckery.

Reply:
well i m new to byond, and i wanted UpdateInfo() to only fire once, at roundstart

Reply:
and after looking through Secret recipe. and asking a maintainer they said it could be done by ssticker. so i did it

Reply:
> Why is this over SSticker? Just do this the normal way without any callbacks or other fuckery.

if you have a better solution , please create a suggestion
coz i dont know of a better solution , as this code is also used in secret sauce